packages/swapper/src/swappers/thorchain/getThorTradeQuote/getTradeQuote.ts

```diff
@@ -17,63 +20,24 @@ export const getThorTradeQuote = async ({
   deps: ThorchainSwapperDeps
   input: GetTradeQuoteInput
 }): Promise<TradeQuote<ChainId>> => {
-  const { sellAsset, buyAsset, sellAmount, sellAssetAccountNumber, wallet } = input
+  const { sellAsset, buyAsset, sellAmount, sellAssetAccountNumber, wallet, chainId } = input
 
   if (!wallet)
     throw new SwapError('[getTradeQuote] - wallet is required', {
       code: SwapErrorTypes.VALIDATION_FAILED
     })
 
   try {
-    const {
-      assetReference: sellAssetErc20Address,
-      chainReference,
-      chainNamespace
-    } = fromAssetId(sellAsset.assetId)
+    const { assetReference: sellAssetErc20Address, assetNamespace } = fromAssetId(sellAsset.assetId)
 
-    const isErc20Trade = sellAssetErc20Address.startsWith('0x')
-    const sellAssetChainId = toChainId({ chainReference, chainNamespace })
-    const adapter = deps.adapterManager.get(sellAssetChainId)
+    const isErc20Trade = assetNamespace === 'erc20'
```

Comment:
Similar story with passing `isErc20Trade`, we have the sellAsset being passed to `getEthThorTxInfo()` and can easily be derived within there. There are just a lot of input args for some of these functions and it would be nice to reduce that input payload a bit if possible. These are more overarching idea than directly related to your PR as it stands, so they don't need to be addressed here, but something to think about

Reply:
for sure we can probably clan this up a bit with regards to passing some things (bip44 instead of account number etc)